src/vs/workbench/contrib/terminal/browser/xterm/quickFixAddon.ts

```diff
@@ -46,15 +48,13 @@ export class TerminalQuickFixAddon extends Disposable implements ITerminalAddon,
 
 	private _terminal: Terminal | undefined;
 
-	private _currentQuickFixElement: HTMLElement | undefined;
-
-	private _decorationMarkerIds = new Set<number>();
-
 	private _commandListeners: Map<string, ITerminalQuickFixOptions[]> = new Map();
 
-	private _quickFixes: IAction[] | undefined;
+	private _quickFixes: Map<number, IAction[]> = new Map();
+	private _decorations: Map<number, IDecoration> = new Map();
```

Comment:
I don't think we need the maps if we're doing it the simpler way?

Reply:
we do and it's not totally clear why - spent awhile yesterday looking into it and didn't work at all when I went back to just a single decoration

Reply:
We only need to track a single list of quick fixes now though right? Same for a single decoration? Removing the maps should greatly simplify things.